doc/arch/adr-009-elasticsearch6-upgrade.md

```diff
@@ -0,0 +1,373 @@
+ES6_TARGET_PORT = os.getenv('ES6_TARGET_HOST', 'http://elasticsearch6:80')
+
+es_client5 = Elasticsearch5([ES5_HOST_PORT])
+es_client6 = Elasticsearch6([ES6_TARGET_PORT])
+
+
+def _prepare_docs_for_bulk_insert(docs):
+    for doc in docs:
+        yield {
+            "_id": doc['_id'],
+            "_source": doc['_source'],
+        }
+
+def bulk_index_documents_to_es6(documents):
+    try:
+        bulk(
+            es_client6,
+            _prepare_docs_for_bulk_insert(documents),
+            index=INDEX,
+            doc_type=GENERIC_DOC_TYPE,
+            chunk_size=100
+        )
+    except TransportError6 as e:
+        print("Failed to index documents: %s", str(e))
+
+
+def fetch_documents(from_=0, page_size=100, scroll_id=None):
+    try:
+        if scroll_id is None:
+            results = es_client5.search(INDEX, GENERIC_DOC_TYPE, from_=from_, size=page_size, scroll='2m')
+            scroll_id = results['_scroll_id']
+        else:
+            results = es_client5.scroll(scroll_id=scroll_id, scroll='2m')
+        docs = results['hits']['hits']
+        return (scroll_id, docs)
+    except TransportError5 as e:
+        print("Failed to fetch documents: %s", str(e))
+        return str(e), e.status_code
+
+
+if __name__ == '__main__':
+    start = datetime.now()
+
+    dcount = es_client5.count(index=INDEX, doc_type=GENERIC_DOC_TYPE)['count']
+
+    print('Preparing to index {} document(s) from ES5'.format(dcount))
+
+    offset = 0
+    page_size = 250
+    scroll_id = None
+    while offset <= dcount:
+        scroll_id, docs = fetch_documents(from_=offset, page_size=page_size, scroll_id=scroll_id)
+
+        print('Indexing documents {} to {} into ES6'.format(offset, offset+page_size))
+        bulk_index_documents_to_es6(docs)
+
+        offset += page_size
+
+    print('Finished in {} seconds'.format(datetime.now() - start))
+```
+
+[search-admin]: https://github.com/alphagov/search-admin
+
+
+## Data sync for production -> staging -> integration
+
+This was done in the same way [as for Elasticsearch 5][].
+
+The data sync script needed to be modified to allow for a different
+host to be used (`http://elasticsearch6` vs `http://elasticsearch5`)
+but otherwise this was just a matter of writing some more
+configuration.
+
+[as for Elasticsearch 5]: https://github.com/alphagov/search-api/blob/master/doc/arch/adr-008-elasticsearch5-upgrade.md
+
+
+## A/B testing Elasticsearch 6
+
+The A/B test was set up like so:
+
+1. Configuration in govuk-cdn-config.
+2. Logic in finder-frontend to pass one of two URL parameters to
+   search-api, based on the CDN-level A/B test.
+3. Logic in search-api to choose a cluster based on the parameter from
+   finder-frontend.
+
+The general A/B test process [is covered in the dev docs][].
+
+For the A/B test we monitored click-through rate, proportion of search
+refinements, and proportion of search exits.  The test revealed a
+significant degradation in search performance compared with
```

Comment:
This could specifically mention relevancy being the issue since performance suggests latency and other metrics.